numpy/_core/src/npysort/quicksort.cpp

```diff
@@ -83,12 +85,14 @@ inline bool quicksort_dispatch(T *start, npy_intp num)
         #endif
         NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::qsort_simd::template QSort, <TF>);
     }
+    #if !DISABLE_HIGHWAY_OPTIMIZATION
     else if (sizeof(T) == sizeof(uint32_t) || sizeof(T) == sizeof(uint64_t)) {
         #ifndef NPY_DISABLE_OPTIMIZATION
             #include "simd_qsort.dispatch.h"
         #endif
         NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::qsort_simd::template QSort, <TF>);
     }
+    #endif
```

Comment:
```suggestion
    else if (sizeof(T) == sizeof(uint32_t) || sizeof(T) == sizeof(uint64_t)) {
        #ifndef NPY_DISABLE_OPTIMIZATION
            #include "simd_qsort.dispatch.h"
        #endif
        NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::qsort_simd::template QSort, <TF>);
        if (dispfunc != nullptr) {
            // Priority is given to Intel-sort library to its efficient support for AVX512.
            // For other CPU targets, we fallback to Google's highway sort.
            #ifndef NPY_DISABLE_OPTIMIZATION
                #include "highway_qsort.dispatch.h"
            #endif
            NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::highway_qsort::template QSort, <TF>);
        }
    }
    #endif
```
Using Google Highway after confirming the unavailability of the Intel library, either due to architectural differences or the AVX512 features not being available during runtime.
Also, note that there is no need for the guard `DISABLE_HIGHWAY_OPTIMIZATION` as the targeted features are controlled through Meson